py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,460 @@
+
+        flags = tools.argparser.parse_args(args=[])
+        store = Storage(self.auth_token)
+        credentials = store.get()
+
+        if not credentials or credentials.invalid:
+            try:
+                flow = client.flow_from_clientsecrets(self.client_secret, SCOPES)
+                flow.user_agent = APPLICATION_NAME
+                if flags:
+                    credentials = tools.run_flow(flow, store, flags)
+                else:  # Needed only for compatibility with Python 2.6
+                    credentials = tools.run(flow, store)
+            except clientsecrets.InvalidClientSecretsError:
+                raise Exception('missing client_secret')
+
+            """
+            Have to restart i3 after getting credentials to prevent bad output.
+            This only has to be done once on the first run of the module.
+            """
+            self.py3.command_output('i3-msg restart')
+
+        return credentials
+
+    def _authorize_credentials(self):
+        """
+        Fetches an access/refresh token by authorizing OAuth 2.0 credentials.
+
+        Returns: True, if the authorization was successful.
+                 False, if a ServerNotFoundError is thrown.
+        """
+        try:
+            http = self.credentials.authorize(httplib2.Http())
+            self.service = discovery.build('calendar', 'v3', http=http)
+            return True
+        except ServerNotFoundError:
+            return False
+
+    def _get_events(self):
+        """
+        Fetches events from the color into a list.
+
+        Returns: The list of events.
+        """
+        self.last_update = datetime.datetime.now()
+        now = datetime.datetime.utcnow().isoformat() + 'Z'  # 'Z' indicates UTC time
+
+        event_cache = self.events
+
+        try:
+            eventsResult = self.service.events().list(
+                calendarId='primary', timeMin=now, maxResults=self.num_events, singleEvents=True,
+                orderBy='startTime').execute(num_retries=5)
+        except:
+            return event_cache
+
+        return eventsResult.get('items', [])
+
+    def _check_warn_threshold(self, time_to, summary):
+        """
+        Checks if the time until an event starts is less than or equal to the warn_threshold.
+        If it is, it presents a warning with self.py3.notify_user.
+        """
+        if time_to['total_minutes'] <= self.warn_threshold:
+            self.py3.notify_user("Warning: Appointment " + str(summary) +
+                                 " coming up!", 'warning', self.warn_timeout)
+
+    def _gstr_to_date(self, date_str):
+        """ Returns a dateime object from a google calendar date string."""
+        return datetime.datetime.strptime(date_str, '%Y-%m-%d')
+
+    def _gstr_to_datetime(self, date_time_str):
+        """ Returns a datetime object from a google calendar date/time string."""
+        tmp = '-'.join(date_time_str.split('-')[:-1])
```

Comment:
For me this function fails since this is converted to `2017-10` and then the format below does not match.

To make it work, I have to change this function to:

```python
    def _gstr_to_datetime(self, date_time_str):
        """ Returns a datetime object from a google calendar date/time string."""
        tmp = '-'.join(date_time_str.split('+')[:-1])
        return datetime.datetime.strptime(tmp, '%Y-%m-%dT%H:%M:%S')
```

Reply:
Sorry about that, I didn't consider positive offsets. That was dumb. Anyways, I took @lasers' suggestion and the module works with UTC offsets now, so that shouldn't be an issue anymore.